org.adempiere.base/src/org/compiere/model/MMeasureCalc.java

```diff
@@ -143,7 +143,7 @@ public String getSqlPI (MGoalRestriction[] restrictions,
 	{
 		StringBuilder sb = new StringBuilder(getSelectClause())
 			.append(" ")
-			.append(getWhereClause());
+			.append("WHERE " + getWhereClause());

```

Comment:
It is unlikely we will accept this - a small convenience that will break backward compatibility.

Reply:
this is breaking change. we migrate GW data, we can add to release notes.
not sure too much people used goals at all - because performance and bugs. i assume this will have small impact. wdyt ?

Reply:
I don't think it is a good idea to break backward compatibility for such small gain. We can wait for @CarlosRuiz-globalqss  opinion on this but it is NO for me.

Reply:
what we see as problem:
we have multiple "engines" to generate data for charts, goals, status lines each of them has a bit another sql definition logic and parsing. this is one of the example where in charts we haven't  WHERE term in where field and goal has. then developer/consultant must remember different writing practice.  We would like a bit standardise with this change.

there are more issues (limitation to write modern/effective sql queries). but this is out of this PR, i will sum-up this on another forum.